docs: deprecate backups with shell scripts

davidekete opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->


### Description

This PR closes the CODA issue: https://github.com/canonical/open-documentation-academy/issues/171

---

### Related Issue

If this pull request addresses an existing issue, please link to it below. Use the `Fixes #<issue-number>` syntax to
[close the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue) automatically when the pull request is merged.

Example:

- Fixes: https://github.com/canonical/open-documentation-academy/issues/171

---

### Contributor License Agreement (CLA)

By contributing to this project, you agree to the terms of
the [Canonical Contributor License Agreement (CLA)](https://ubuntu.com/legal/contributors).  
If you have not already signed the CLA, [please do so here](https://ubuntu.com/legal/contributors).

---

### Commit Message for Squash Merge

We typically squash commits when merging. You can specify the commit message that should be used in this case if you wish.
Provide the desired commit message below:

[(optional) category] Brief description of changes made, and why

---

### Checklist

- [x] I have read and followed the [Ubuntu Server contributing guide](https://documentation.ubuntu.com/server/contributing/).
- [x] I have signed the [Contributor License Agreement (CLA)](https://ubuntu.com/legal/contributors).
- [x] My pull request is linked to an existing issue (if applicable).
- [x] I have tested my changes, and they work as expected.

---

### Additional Notes (Optional)

Add any extra information or context that reviewers may need to know. This could include testing instructions,
screenshots, or links to related discussions.

---

Thank you for contributing to the Ubuntu Server documentation!

s-makin reviewed (changes requested):
Thanks @davidekete for submitting this PR. It will need a couple of changes before I can merge it -- the pages that used to refer to the deleted page need their links to be changed, rather than removed.

The RTD preview still builds (I guess because of the `"deprecate-\"backup-with-shell-scripts\""` message) but doing the deprecation this way means that the links are just removed, without any changes to the text. This would degrade the user experience, if they come across text like [in the how-to page](https://documentation.ubuntu.com/server/how-to/backups/back-up-using-shell-scripts/) that tells them to refer to the example reference page -- which they won't be able to find, because the link is removed, but so is the page. It's much better to point to the correct place to make life easier for the user.

Instead of using this deprecation method, I would strongly prefer that we replace each link that *would* be broken with the URL to the Discourse post. The surrounding text may also need to be changed to make it make sense (since it will now point outside of the docs). This will need to be done on:
* /explanation/intro-to/backups.md (line 36)
* /how-to/backups/back-up-using-shell-scripts.md (line 12)
* /reference/backups/archive-rotation-shell-script.md (line 5)

We will also need a permanent redirect set up [in the custom_conf.py](https://github.com/canonical/ubuntu-server-documentation/blob/7b909807a727d1dec85630a30282478f89ad9b80/custom_conf.py#L130) file to ensure that any traffic coming in via users' bookmarks gets sent to the Discourse post URL as well.

davidekete commented:
Hi @s-makin, thanks for the review. In my last commit, I:

- Set up a permanent redirect in the custom_conf.py
- Changed the surrounding text and fixed the broken links on:
  - /explanation/intro-to/backups.md (line 36)
  - /how-to/backups/back-up-using-shell-scripts.md (line 12)
  - /reference/backups/archive-rotation-shell-script.md (line 5)

davidekete commented:
@s-makin bump

s-makin commented on how-to/backups/back-up-using-shell-scripts.md:
```diff
@@ -9,7 +9,7 @@ In this guide, we will walk through how to use a shell script for backing up fil
 ## The shell script
 
-The following shell script uses the basic backup shell script from our Reference section. It uses `tar` to create an archive file on a remotely mounted NFS file system. The archive filename is determined using additional command line utilities. For more details about the script, check out the {ref}`example Reference page <basic-backup-shell-script>`.
+The following shell script, which is now deprecated uses `tar` to create an archive file on a remotely mounted NFS file system. The archive filename is determined using additional command line utilities. For more details about the script, check out the basic backup shell script <https://discourse.ubuntu.com/t/basic-backup-shell-script/36419>`_.
```
```suggestion
The following shell script uses `tar` to create an archive file on a remotely mounted NFS file system. The archive filename is determined using additional command line utilities. For more details about the script, check out the `basic backup shell script <https://discourse.ubuntu.com/t/basic-backup-shell-script/36419>`_.
```

s-makin commented on redirects.txt:
```diff
@@ -471,7 +471,7 @@ cloud-images/introduction reference/clouds/find-cloud-images/
 migrate-from-crmsh-to-pcs reference/high-availability/migrate-from-crmsh-to-pcs/
 
-basic-backup-shell-script reference/backups/basic-backup-shell-script/
+# basic-backup-shell-script reference/backups/basic-backup-shell-script/
```
```suggestion
basic-backup-shell-script reference/backups/basic-backup-shell-script/
```

s-makin replied:
This is an old redirect that used to exist in Discourse, which we need to keep. The "to" path is correctly pointing back to Discourse, but we can't take it out because otherwise users on the old path will 404, and I want to send them back to the Discourse post (it's a very confusing circle, but necessary to maintain)

s-makin commented on reference/backups/archive-rotation-shell-script.md:
````diff
@@ -82,7 +82,7 @@ date
 ```
 
-The script can be executed using the same methods as in our guide on {ref}`how to back up using shell scripts <back-up-using-shell-scripts>`.
+The script can be executed using the same methods as in this basic backup shell script <https://discourse.ubuntu.com/t/basic-backup-shell-script/36419>`_.
````
```suggestion
The script can be executed using the same methods as in this `basic backup shell script <https://discourse.ubuntu.com/t/basic-backup-shell-script/36419>`_.
```

s-makin commented on reference/backups.rst:
```diff
@@ -5,11 +5,10 @@ Backups and version control
 Example shell scripts for backing up your system.
 
-* :ref:`Basic backup shell script <basic-backup-shell-script>`
+* [DEPRECATED] `Basic backup shell script <https://discourse.ubuntu.com/t/basic-backup-shell-script/36419>`_
```
```suggestion
```

s-makin replied:
It's fine to remove this one - the redirects will handle this case.

s-makin reviewed (commented):
Sorry for the delay @davidekete, I got interrupted mid-review and forgot to submit it!

There are a couple of fixups to the links, which I've added suggestions for. I do have a question about the banner I'm seeing on the front page of the PR:

![image](https://github.com/user-attachments/assets/97ddb572-0166-4432-ac18-1a5d09200f13)

I didn't see this text anywhere in any of the changed files, and I can't figure out why it's still showing up as a warning. I'm happy to apply the fixups, but would you mind looking into this and undoing whatever it is that's feeding that warning?

davidekete commented:
> Sorry for the delay @davidekete, I got interrupted mid-review and forgot to submit it!
> 
> There are a couple of fixups to the links, which I've added suggestions for. I do have a question about the banner I'm seeing on the front page of the PR:
> 
> ![image](https://private-user-images.githubusercontent.com/116022293/422895175-97ddb572-0166-4432-ac18-1a5d09200f13.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIwMDYyOTMsIm5iZiI6MTc0MjAwNTk5MywicGF0aCI6Ii8xMTYwMjIyOTMvNDIyODk1MTc1LTk3ZGRiNTcyLTAxNjYtNDQzMi1hYzE4LTFhNWQwOTIwMGYxMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxNVQwMjMzMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YjU2ODAxMTMxNDhkMzk0ZTAzNmFiZTQwZWYyNDJhMjgxYTRkZjZkNDBiYjQ2NTUyN2M1ZWM3YzQwMGExNjVhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1MpbnAxCY2KPG9PS0l9di2_6kcFlSuEOTXG9VL7Slps)
> 
> I didn't see this text anywhere in any of the changed files, and I can't figure out why it's still showing up as a warning. I'm happy to apply the fixups, but would you mind looking into this and undoing whatever it is that's feeding that warning?

Hi @s-makin, that warning occurs because the branch name literally contains special or “hidden” characters—in this case, the backslash-escaped quotes (my bad). It's GitHub's way of warning you that the name might not be what it appears at first glance. Renaming the branch would make the error go away. However, I can't rename the branch without closing this PR and opening a new one which I do not recommend.

s-makin commented:
> > Sorry for the delay @davidekete, I got interrupted mid-review and forgot to submit it!
> > There are a couple of fixups to the links, which I've added suggestions for. I do have a question about the banner I'm seeing on the front page of the PR:
> > ![image](https://private-user-images.githubusercontent.com/116022293/422895175-97ddb572-0166-4432-ac18-1a5d09200f13.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIwMDYyOTMsIm5iZiI6MTc0MjAwNTk5MywicGF0aCI6Ii8xMTYwMjIyOTMvNDIyODk1MTc1LTk3ZGRiNTcyLTAxNjYtNDQzMi1hYzE4LTFhNWQwOTIwMGYxMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxNVQwMjMzMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YjU2ODAxMTMxNDhkMzk0ZTAzNmFiZTQwZWYyNDJhMjgxYTRkZjZkNDBiYjQ2NTUyN2M1ZWM3YzQwMGExNjVhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1MpbnAxCY2KPG9PS0l9di2_6kcFlSuEOTXG9VL7Slps)
> > I didn't see this text anywhere in any of the changed files, and I can't figure out why it's still showing up as a warning. I'm happy to apply the fixups, but would you mind looking into this and undoing whatever it is that's feeding that warning?
> 
> Hi @s-makin, that warning occurs because the branch name literally contains special or “hidden” characters—in this case, the backslash-escaped quotes (my bad). It's GitHub's way of warning you that the name might not be what it appears at first glance. Renaming the branch would make the error go away. However, I can't rename the branch without closing this PR and opening a new one which I do not recommend.

Ahhh, yes, that makes sense! Thanks for clarifying - I agree, there's no point in redoing the work on another branch, I'm happy to merge this as-is. Thanks again for all your work on this issue!
